mathesar_ui/src/component-library/types.d.ts

```diff
@@ -1,9 +1,18 @@
+import type { SvelteComponent } from 'svelte';
+
 export type Appearance = 'default' | 'primary' | 'secondary' | 'plain' | 'ghost';
 export type Size = 'small' | 'medium' | 'large';
 
+export interface Option {
+  value: string | number,
+  label?: string,
+  labelComponent?: typeof SvelteComponent,
+  labelComponentProps?: unknown,
+  disabled?: boolean,
+}
+
```

Comment:
I find it very useful in several circumstances having a flexible option structure for Select as well as any component that relies on an array which contains a value and a label (It is fine to have a rigid structure for other properties like disabled and labelComponent since they directly correspond to presentation part of things).

Moreoever, this is a feature almost every popular component library provides.

I understand you like a rigid structure better because it helps with better typing. As a good solution, we can provide a sensible BaseOption type and an Option type. The BaseOption type would be less rigid while Option extends it and would be more rigid.